src/olympia/shelves/forms.py

```diff
@@ -38,25 +40,38 @@ def clean(self):
         if criteria is None:
             return
 
-        if endpoint == 'search':
+        if endpoint in (Shelf.Endpoints.SEARCH, Shelf.Endpoints.TAGS):
             if not criteria.startswith('?') or criteria.count('?') > 1:
                 raise forms.ValidationError('Check criteria field.')
-            params = criteria[1:].split('&')
-            if addon_type == amo.ADDON_EXTENSION and 'type=statictheme' in params:
+            params = dict(parse.parse_qsl(criteria.strip('?')))
+            if (
+                addon_type == amo.ADDON_EXTENSION
+                and params.get('type') == 'statictheme'
+            ):
                 raise forms.ValidationError(
                     'Use "Theme (Static)" in Addon type field for type=statictheme.'
                 )
             elif (
-                addon_type == amo.ADDON_STATICTHEME and 'type=statictheme' not in params
+                addon_type == amo.ADDON_STATICTHEME
+                and params.get('type') != 'statictheme'
             ):
                 raise forms.ValidationError(
                     'Check fields - for "Theme (Static)" addon type, use '
                     'type=statictheme. For non theme addons, use "Extension" in Addon '
                     'type field, not "Theme (Static)".'
                 )
+            if endpoint == Shelf.Endpoints.TAGS:
+                if 'tag' in params:
+                    raise forms.ValidationError(
+                        'Omit `tag` param for tags shelf - a random tag will be chosen.'
```

Comment:
Would it be worth allowing for a `tag` to be specified? Or is that use case dealt with by simply using the `search` endpoint?

Reply:
It'd make more sense to use the search endpoint, at least as it's currently written (the _only_ additional functionality is the random tag value so if you're overriding that it's pointless).  That might change depending on how the other issues end up being implemented, but I'll change/drop the error then.